core/runtime/src/main/java/io/quarkus/runtime/Startup.java

```diff
@@ -12,11 +12,20 @@
 import javax.enterprise.inject.spi.ObserverMethod;
 
 /**
- * This annotation can be used to initialize a CDI bean at application startup. The behavior is similar to a declaration of an
- * observer of the {@link StartupEvent} - a contextual instance is created and lifecycle callbacks (such as
- * {@link javax.annotation.PostConstruct}) are invoked. In fact, a synthetic observer of the {@link StartupEvent} is generated
- * for each bean annotated with this annotation. Furthermore, {@link #value()} can be used to specify the priority of the
- * generated observer method and thus affect observers ordering.
+ * This annotation can be used to initialize a CDI bean at application startup:
+ * <ul>
+ * <li>If a bean class is annotated then a contextual instance is created and the {@link javax.annotation.PostConstruct}
+ * callbacks are invoked.</li>
+ * <li>If a producer method is annotated then a contextual instance is created, i.e. the producer method is invoked.</li>
+ * <li>If a producer field is annotated then a contextual instance is created, i.e. the producer field is read.</li>
+ * <li>If a non-static non-producer no-args method of a bean class is annotated then a contextual instance is created, the
+ * lifecycle callbacks are invoked and finally the method itself is invoked.</li>
```

Comment:
This should describe what happens if the bean has multiple `@Startup`-annotated methods (they are invoked in the priority order, I assume, and all on the same instance?).

Reply:
Also, can a `@Startup` method be inherited? :-)

Reply:
Fair enough re the ordering -- I only noticed the explanation of synthetic observers after I wrote this comment.

> > and all on the same instance?).
> 
> It depends on the scope of the bean.

Now this is interesting. If the bean is `@Dependent`, each `@Startup` method will be invoked on a fresh instance?

I guess that follows from how observer methods work, so, yea...

Reply:
> I guess that follows from how observer methods work, so, yea...

Exactly.

Reply:
> Also, can a `@Startup` method be inherited? :-)

No, it can't. No inheritance here. And it's not even annotated with `@Inherited`.

Reply:
Gotcha. It's just a different syntax for defining an observer method, and hence works identically to observers. The javadoc kinda sorta states that, so it should be fine.